component/async/amqp/integration_test.go

```diff
@@ -0,0 +1,192 @@
+// +build integration
+
+package amqp
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/beatlabs/patron/encoding/json"
+	"github.com/streadway/amqp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestConsumeAndPublish(t *testing.T) {
+	// Setup consumer.
+	f := &Factory{
+		url:      "amqp://guest:guest@localhost/",
+		queue:    "async-amqp-queue",
+		exchange: *validExch,
+		bindings: []string{},
+	}
+	c, err := f.Create()
+	require.NoErrorf(t, err, "failed to create consumer: %v", err)
+	ctx := context.Background()
+	msgChan, errChan, err := c.Consume(ctx)
+	assert.NotNil(t, msgChan)
+	assert.NotNil(t, errChan)
+	assert.NoError(t, err)
+
+	conn, ch := setupRabbitMQPublisher(t)
+	defer func() {
+		err := conn.Close()
+		assert.NoError(t, err)
+	}()
+
+	// Wait for everything to be set up properly.
+	time.Sleep(2000 * time.Millisecond)
```

Comment:
Can we avoid the sleep ?
It would be better, if we could have a polling strategy to check if everything is set up ... and have a timeout with a ticker maybe ? What do you think ?

Reply:
This was the simplest solution to deal with fragile tests. We could poll the publisher and check whether the queue has been declared properly before sending messages. I'll make something up and we'll see how it looks.